src/app/services/courses.service.ts

```diff
@@ -1,57 +1,73 @@
 import { Injectable } from '@angular/core';
-import COURSES from '../models/mock-courses';
-import { CourseInfo } from '../models/interfaces';
 import { Router } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
+import { Observable, Subject } from 'rxjs';
+
+import { CourseInfo } from '../models/interfaces';
+import { SERVER_URL } from '../models/const';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CoursesService {
-  private coursesList: CourseInfo[] = COURSES.slice();
+  public searchFilterUpdated$: Subject<boolean> = new Subject();
 
-  constructor(private router: Router) { }
+  private count = 4;
+  private filter = '';
 
-  public getCourses() {
-    return this.coursesList.slice();
+  constructor(
+    private router: Router,
+    private http: HttpClient
+  ) { }
+
+  public getCourses(start: number): Observable<CourseInfo[]> {
+    const url = SERVER_URL + `/courses?_start=${start}&_limit=${this.count}&q=${this.filter}`;
+
+    return this.http.get<CourseInfo[]>(url);
   }
 
-  public getCourseById(id: number): CourseInfo {
-    return this.coursesList.filter((c) => c.id === id)[0];
+  public getCourseById(id: number): Observable<CourseInfo> {
+    const url = SERVER_URL + `/courses/${id}`;
+
+    return this.http.get<CourseInfo>(url);
   }
 
-  public getCourseByName(name: string): CourseInfo {
-    return this.coursesList.filter((c) => c.title === name)[0];
+  public getCourseByName(title: string): Observable<CourseInfo[]> {
+    const url = SERVER_URL + `/courses?title=${title}`;
+
+    return this.http.get<CourseInfo[]>(url);
   }
 
   public createCourse(course: CourseInfo) {
-    this.coursesList.push(course);
-
-    this.router.navigate(['/Courses']);
+    const url = SERVER_URL + '/courses';
 
-    return course.id;
+    return this.http.post<CourseInfo>(url, course)
+      .subscribe(() => {
+        this.router.navigate(['/Courses']);
```

Comment:
It's better to keep this logic in component for now, but later it will be mooved to the NgRX effects. Basically, service that works with http should do only work with http and shouldn't do anything else.

Reply:
Moved to component